src/uct/ib/ud/base/ud_iface.c

```diff
@@ -68,11 +71,20 @@ uct_ud_iface_cep_get_conn_sn(uct_ud_iface_t *iface,
                              int path_index)
 {
     void *peer_address = ucs_alloca(iface->conn_match_ctx.address_length);
-    return (uct_ud_ep_conn_sn_t)
-           ucs_conn_match_get_next_sn(&iface->conn_match_ctx,
-                                      uct_ud_iface_cep_get_peer_address(
-                                          iface, ib_addr, if_addr, path_index,
-                                          peer_address));
+    uct_ud_ep_conn_sn_t conn_sn;
+
+    conn_sn =
+            (uct_ud_ep_conn_sn_t)
+            ucs_conn_match_get_next_sn(&iface->conn_match_ctx,
+                                       uct_ud_iface_cep_get_peer_address(
+                                               iface, ib_addr, if_addr,
+                                               path_index, peer_address));
+    if (conn_sn == UCT_UD_IFACE_CEP_CONN_SN_MAX) {
+        ucs_fatal("connection ID reached the maximal possible value: %u",
+                  UCT_UD_IFACE_CEP_CONN_SN_MAX);
+    }
```

Comment:
why need this assert?

Reply:
UD transport doesn't support case when connection matching is not enabled

Reply:
can we assert that conn match insert in ud always returns 1?

Reply:
done